src/audio/mixin_mixout/mixin_mixout.c

```diff
@@ -190,7 +190,9 @@ static struct comp_dev *mixout_new(const struct comp_driver *drv,
 
 	dev->ipc_config = *config;
 
-	md = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM, sizeof(*md));
+	/* align the allocation size to a cache line for the coherent API */
+	md = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM,
+		     ALIGN_UP(sizeof(*md), PLATFORM_DCACHE_ALIGN));
```

Comment:
@kv2019i no, the alignment that those allocations do (as well as all our allocations) is of the start address. But we also need to align the end address to avoid a part of this allocation sharing a cache-line with a header of the next allocation. This is done with cached allocation, check in alloc.c. But since here we allocate uncached, we align the size manually.

Reply:
@lyakh Hmm, unless SOF_MEM_FLAG_COHERENT is passed, SOF_MEM_ZONE_RUNTIME allocs are always cached, and heap_alloc_aligned_cached specifically rounds up the allocation size. Otherwise we'd have issues with Zephyr heap metadata.

Reply:
@kv2019i https://github.com/thesofproject/sof/blob/73f6171bb3c8c5b1d0963883014508022445b997/src/audio/buffer.c#L44 allocates as `SHARED` i.e. uncached, right? So that one won't be aligned up. As for mixin-mixout - I think I took the wrong allocation :-/ It's https://github.com/thesofproject/sof/blob/73f6171bb3c8c5b1d0963883014508022445b997/src/audio/mixin_mixout/mixin_mixout.c#L201-L202 that has to be changed - that's the allocation, used with the coherent API... I'll change that... Does that sound correct?

Reply:
@lyakh Ack, _SHARED is uncached.